File: trees/1-level_order_traveresal.java
//leet code 102

class Solution {
    public List<List<Integer>> levelOrder(TreeNode node) {
        
        
      List<List<Integer>> al = new ArrayList<>();
        if(node==null) return al;
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        TreeNode root = node;
        q.add(root);
        
        while(!q.isEmpty())
        {
            int c= q.size();
             List<Integer> li = new ArrayList<>();
            for(int i=0;i<c;i++)
            {

            TreeNode temp =q.poll();
            li.add(temp.val);
                
            // System.out.print(temp.val+ " ");
            
            if(temp.left!=null)
            q.add(temp.left);
            
            if(temp.right!=null)
            q.add(temp.right);
            }
            al.add(li);
               
        }
       
        return al;  
    }
}
// gfg

class Level_order_traversal
{
    //You are required to complete this method
    static ArrayList <Integer> levelOrder(Node node) 
    {
       ArrayList <Integer> al = new ArrayList<>();
        Queue<Node> q = new LinkedList<Node>();
        Node root = node;
        q.add(root);
        
        while(!q.isEmpty())
        {
            Node temp =q.poll();
            al.add(temp.data);
            
            if(temp.left!=null)
            q.add(temp.left);
            
            if(temp.right!=null)
            q.add(temp.right);
            
            
           
           
        }
       
        return al;   
    }
}

File: trees/1-level_order_traveresal.cpp
//leetcode 102

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* node) {
        vector<vector<int>> ans;
        if(node==NULL) return ans;
        queue<TreeNode*> q;
        TreeNode* root = node;
        q.push(root);
        
        while(!q.empty()) {
            int c = q.size();
            vector<int> li;
            for(int i=0; i<c; i++) {
                TreeNode* temp = q.front();
                q.pop();
                li.push_back(temp->val);
                if(temp->left != NULL)
                    q.push(temp->left);
                if(temp->right != NULL)
                    q.push(temp->right);
            }
            ans.push_back(li);
        }
       
        return ans;  
    }
};

//gfg

class Solution
{
    public:
    //Function to return the level order traversal of a tree.
    vector<int> levelOrder(Node* node)
    {
      //Your code her
       vector<int> result;
        queue<Node*> q;
        Node* root = node;
        q.push(root);
        
        while(!q.empty()) {
            Node* temp = q.front();
            q.pop();
            result.push_back(temp->data);
            if(temp->left != nullptr) q.push(temp->left);
            if(temp->right != nullptr) q.push(temp->right);
        }
        return result;
    }
};